integration/specs/Picklist/picklist-1.spec.js

```diff
@@ -0,0 +1,116 @@
+        browser.keys(SPACE_KEY);
+        picklist.waitUntilOpen();
+        expect(picklist.isMenuOpen()).toBe(true);
+        expect(picklist.hasFocusInput()).toBe(true);
+    });
+    it('should close menu and not lose input focus when press esc key', () => {
+        const picklist = new PagePicklist(PICKLIST);
+        picklist.clickInput();
+        picklist.waitUntilOpen();
+        browser.keys(ESCAPE_KEY);
+        expect(picklist.isMenuOpen()).toBe(false);
+        expect(picklist.hasFocusInput()).toBe(true);
+    });
+    it('should close menu and lose input focus when press tab key', () => {
+        const picklist = new PagePicklist(PICKLIST);
+        picklist.clickInput();
+        picklist.waitUntilOpen();
+        browser.keys(TAB_KEY);
+        expect(picklist.isMenuOpen()).toBe(false);
+        expect(picklist.hasFocusInput()).toBe(false);
+    });
+    it('should close menu when click outside it', () => {
+        const picklist = new PagePicklist(PICKLIST);
+        const logoElement = $(REACT_LOGO);
+        picklist.clickInput();
+        picklist.waitUntilOpen();
+        logoElement.click();
+        expect(picklist.isMenuOpen()).toBe(false);
+    });
+    it("should select 'Empire State' with keyboard", () => {
+        const picklist = new PagePicklist(PICKLIST);
+        picklist.clickInput();
+        picklist.waitUntilOpen();
+        browser.keys(ARROW_DOWN_KEY);
+        browser.keys(ENTER_KEY);
+        expect(picklist.getLabel()).toBe('Empire State');
+    });
+    it("should set active 'Empire State' when hover the option", () => {
+        const picklist = new PagePicklist(PICKLIST);
+        picklist.clickInput();
+        picklist.waitUntilOpen();
+        const option = picklist.getOption(1);
+        option.hover();
+        expect(option.isActive()).toBe(true);
+    });
+    it('should set active the first option when other is active and close the menu and open it again', () => {
+        const picklist = new PagePicklist(PICKLIST);
+        picklist.clickInput();
+        picklist.waitUntilOpen();
+        let option = picklist.getOption(0);
+        expect(option.isActive()).toBe(true);
+        browser.keys(ARROW_DOWN_KEY);
+        browser.keys(ENTER_KEY);
+        browser.keys(ARROW_DOWN_KEY);
+        picklist.waitUntilOpen();
+        option = picklist.getOption(0);
+        expect(option.isActive()).toBe(true);
+    });
+    it('should hide the option selected', () => {
+        const picklist = new PagePicklist(PICKLIST);
+        picklist.clickInput();
+        picklist.waitUntilOpen();
+        const option = picklist.getOption(0);
+        const optionLabel = option.getLabel();
+        browser.keys(ENTER_KEY);
+        const labels = [];
+        for (let index = 0; index < picklist.getRegisteredOptionsLength(); index += 1) {
+            const opt = picklist.getOption(index);
+            labels.push(opt.getLabel());
+        }
+        expect(labels.includes(optionLabel)).toBe(false);
```

Comment:
instead of iterate over all options check that the selected option in not visible with the isVisible method created in PicklistOption page object
in this case you need to open menu, select an option, open menu again and check that the selected option is not visible